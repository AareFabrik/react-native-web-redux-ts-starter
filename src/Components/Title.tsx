import * as React from "react";
import { StyleSheet, Text, View, ViewStyle, TextStyle } from "react-native";

export interface Props {
  title: string;
}

export interface State {}

export default class Title extends React.Component<Props, State> {
  constructor(props) {
    super(props);
    this.state = {};
  }
  render() {
    let { title } = this.props;
    return (
      <View style={styles.container}>
        <Text style={styles.title}>{title}</Text>
      </View>
    );
  }
}

interface Style {
  container: ViewStyle;
  title: TextStyle;
}

const styles = StyleSheet.create<Style>({
  container: {
    paddingVertical: 20
  },
  title: {
    color: "#000",
    fontSize: 30,
    fontFamily: "CabinSketch"
  }
});
