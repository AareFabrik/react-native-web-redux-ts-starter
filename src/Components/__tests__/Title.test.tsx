import * as React from "react";
import * as renderer from "react-test-renderer";
import Title from "../Title";

test("Default Title renders correctly", () => {
  const tree = renderer.create(<Title title={"Title"} />).toJSON();
  expect(tree).toMatchSnapshot();
});
