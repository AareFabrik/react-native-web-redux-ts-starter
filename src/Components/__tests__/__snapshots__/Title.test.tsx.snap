// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Title renders correctly 1`] = `
<div
  style={
    Object {
      "WebkitAlignItems": "stretch",
      "WebkitBoxAlign": "stretch",
      "WebkitBoxDirection": "normal",
      "WebkitBoxOrient": "vertical",
      "WebkitFlexBasis": "auto",
      "WebkitFlexDirection": "column",
      "WebkitFlexShrink": 0,
      "alignItems": "stretch",
      "borderBottomStyle": "solid",
      "borderBottomWidth": "0px",
      "borderLeftStyle": "solid",
      "borderLeftWidth": "0px",
      "borderRightStyle": "solid",
      "borderRightWidth": "0px",
      "borderTopStyle": "solid",
      "borderTopWidth": "0px",
      "boxSizing": "border-box",
      "display": "flex",
      "flexBasis": "auto",
      "flexDirection": "column",
      "flexShrink": 0,
      "marginBottom": "0px",
      "marginLeft": "0px",
      "marginRight": "0px",
      "marginTop": "0px",
      "minHeight": "0px",
      "minWidth": "0px",
      "msFlexAlign": "stretch",
      "msFlexDirection": "column",
      "msFlexNegative": 0,
      "msFlexPreferredSize": "auto",
      "paddingBottom": "20px",
      "paddingLeft": "0px",
      "paddingRight": "0px",
      "paddingTop": "20px",
      "position": "relative",
      "zIndex": 0,
    }
  }
>
  <div
    dir="auto"
    style={
      Object {
        "borderBottomWidth": "0px",
        "borderLeftWidth": "0px",
        "borderRightWidth": "0px",
        "borderTopWidth": "0px",
        "boxSizing": "border-box",
        "color": "rgba(0,0,0,1.00)",
        "display": "inline",
        "fontFamily": "CabinSketch",
        "fontSize": "30px",
        "fontStyle": "inherit",
        "fontVariant": "inherit",
        "fontWeight": "inherit",
        "lineHeight": "inherit",
        "marginBottom": "0px",
        "marginLeft": "0px",
        "marginRight": "0px",
        "marginTop": "0px",
        "paddingBottom": "0px",
        "paddingLeft": "0px",
        "paddingRight": "0px",
        "paddingTop": "0px",
        "textDecoration": "none",
        "whiteSpace": "pre-wrap",
        "wordWrap": "break-word",
      }
    }
  >
    Title
  </div>
</div>
`;
